lte/gateway/c/oai/tasks/s1ap/s1ap_common.h

```diff
@@ -393,67 +396,41 @@ extern int asn1_xer_print;
 #include "3gpp_33.401.h"
 #include "security_types.h"
 #include "common_types.h"
-
 #include "s1ap_state.h"
 
-// Forward declaration
-struct s1ap_message_s;
+#define S1AP_FIND_PROTOCOLIE_BY_ID(IE_TYPE, ie, container, IE_ID, mandatory)   \
+  do {                                                                         \
+    IE_TYPE** ptr;                                                             \
+    ie = NULL;                                                                 \
+    for (ptr = container->protocolIEs.list.array;                              \
+         ptr < &container->protocolIEs.list                                    \
+                    .array[container->protocolIEs.list.count];                 \
+         ptr++) {                                                              \
+      if ((*ptr)->id == IE_ID) {                                               \
+        ie = *ptr;                                                             \
+        break;                                                                 \
+      }                                                                        \
+    }                                                                          \
+    if (ie == NULL) {                                                          \
+      if (mandatory)                                                           \
+        OAILOG_ERROR(                                                          \
+            LOG_S1AP,                                                          \
+            "S1AP_FIND_PROTOCOLIE_BY_ID: %s %d: Mandatory ie is NULL\n",       \
+            __FILE__, __LINE__);                                               \
+      else                                                                     \
+        OAILOG_DEBUG(                                                          \
+            LOG_S1AP,                                                          \
+            "S1AP_FIND_PROTOCOLIE_BY_ID: %s %d: Optional ie is NULL\n",        \
+            __FILE__, __LINE__);                                               \
+    }                                                                          \
+    if (mandatory) DevAssert(ie != NULL);                                      \
```

Comment:
Remove the DevAssert. The entire process does not have to crash if a mandatory IE is missing. Logging an error and not responding to the message is enough.